lib/core/ensureBlock.js

```diff
@@ -2,15 +2,20 @@ const ZMQClient = require('@dashevo/dashd-zmq');
 
 /**
  *
- * @param {EventEmitter} socketClient
+ * @param {ZMQClient} zmqClient
  * @param {RpcClient} rpcClient
  * @param {string} hash
  * @return {Promise<void>}
  */
-async function ensureBlock(socketClient, rpcClient, hash) {
+async function ensureBlock(zmqClient, rpcClient, hash) {
+  // Because a ChainLock may happen before its block, we also subscribe to rawblock
+  zmqClient.subscribe(ZMQClient.TOPICS.hashblock);
+
   const eventPromise = new Promise((resolve) => {
-    socketClient.on(ZMQClient.TOPICS.hashblock, (response) => {
-      if (hash === undefined || hash === response) {
+    zmqClient.on(ZMQClient.TOPICS.hashblock, (response) => {
+      if (hash === response) {
+        zmqClient.removeAllListeners(ZMQClient.TOPICS.hashblock);
```

Comment:
It could lead to problems when other part of the code that will use zmqClient. Let's just remove this exact listener?